src/sys/socket/mod.rs

```diff
@@ -1015,6 +1020,11 @@ impl ControlMessageOwned {
                 let dl = ptr::read_unaligned(p as *const libc::sockaddr_in6);
                 ControlMessageOwned::Ipv6OrigDstAddr(dl)
             },
+            #[cfg(all(target_os = "linux"))]
```

Comment:
Fine, but be warned I have not tested it on Android at all nor do I plan on maintaining/supporting that platform personally. My general opinion is that it's better to wait for someone who needs it, and then they can be a point of contact re: getting it working on that platform. That person isn't me for Android.

Reply:
I had to revert this, since the libc constants for this are not available on android (my fault, since I did that PR too, but I'm not willing to block this PR on this).

Reply:
Oops.  I thought the constant _was_ available on Android in libc.  I guess I grepped wrong.  Sorry.